### ISSUE TITLE
Breaking of code embedded in raw strings with string interpolation

### ISSUE THREAD

**Issue by pvginkel** [belav/csharpier#1136]
**Input:**

See output.

**Output:**

```cs
var v = $"""
    abcde abcde abcde abcde abcde {string.Join(
        ", ",
        new[] { 1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12, 13, 14, 15 }
    )}
    """;
```

**Expected behavior:**

Not sure :). Maybe this?

```cs
var v = $"""
    abcde abcde abcde abcde abcde {
        string.Join(", ", new[] { 1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12, 13, 14, 15 })
    }
    """;
```

or

```cs
var v = $"""
    abcde abcde abcde abcde abcde {
        string.Join(
            ", ",
            new[] { 1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12, 13, 14, 15 }
        )
    }
    """;
```

I just don't think the current formatting is ideal.

**Comment by belav** [comment]
Right now there is no formatting applied to expressions that appear within a raw interpolated string. 

From what I recall, the original interpolated strings did not allow line breaks in any expressions within them. So CSharpier was forcing those expressions to be flat. When raw interpolated strings were first supported in CSharpier, they were treated the same way which would have resulted in
```c#
var v = $"""
    abcde abcde abcde abcde abcde {string.Join(", ", new[] { 1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12, 13, 14, 15 })}
    """;
```

And then when I was getting raw strings to indent correctly, the side effect was that I removed all formatting from the expressions in them.

Long story short, I believe it should be possible to get expressions within raw strings to format.

**Comment by RReverser** [comment]
Raw string literals, by their multiline nature, are currently the largest unformatted (well, manually formatted) pieces of code in our codebase. I briefly looked into the current implementation but I'm afraid the logic for raw literals is quite beyond my expertise to implement this formatting myself in a PR. @belav is this something still on your radar?

**Comment by OleRoss** [comment]
Ran into this as well. My use case is in a source generator.

Just wanted to say I would prefer no formatting in a (multi line) interpolated raw string. When I use interpolated strings with line breaks, I want to see where those line breaks are located; it gets really confusing otherwise.

I think I prefer
```csharp
string myString = $"""
     aklshdfasd
     asdasdasdasd {myFunc(param.Type)} aaaaaaaaaaaa aaaaaaaaaaaaa aaaaaaaaaaaaaaa
     soasd
     """;
```

over
```csharp
string myString = $"""
    aklshdfasd
    asdasdasdasd {myFunc(
        param.Type
    )} aaaaaaaaaaaa aaaaaaaaaaaaa aaaaaaaaaaaaaaa
    soasd
    """;
```

@belav is there a world where we just dont format multi-line string literals?
